tests/brevitas/hyp_helper.py

```diff
@@ -218,3 +220,21 @@ def min_max_tensor_random_shape_st(draw, min_dims=1, max_dims=4, max_size=3, wid
     min_tensor = torch.tensor(min_list).view(*shape)
     max_tensor = torch.tensor(max_list).view(*shape)
     return min_tensor, max_tensor
+
+
+@st.composite
+def random_minifloat_format(draw, min_bit_width=MIN_INT_BIT_WIDTH, max_bit_with=MAX_INT_BIT_WIDTH):
+    """"
+    Generate a minifloat format. Returns bit_width, exponent, mantissa, and signed.
+    """
+    bit_width = draw(st.integers(min_value=min_bit_width, max_value=max_bit_with))
+    exponent_bit_width = draw(st.integers(min_value=0, max_value=bit_width))
+    signed = draw(st.booleans())
+    # if no budget is left, return
```

Comment:
What does this mean?

Reply:
It means that the exponent_bit_width + signed at that point might already sum up to bit_width. If that's the case, we can't draw mantissa bits anymore, so returning it with a 0 for mantissa_bit_width is the only option

Reply:
No the base case is 1 bit for everything (e.g., 3 bit floating point)

Reply:
Agreed, added the checks for that later, so this is still from the earlier tests.